src/asciidoc/docs/guides-general/build-custom-controls.adoc

```diff
@@ -0,0 +1,210 @@
+
+  // When the custom control has loaded, register it to trigger events from the consumer app.
+  useEffect(() => {
+    myCustomCtrl.register();  // <4>
+  }, []);
+
+  // Call the ready method once (with an empty dependency array)
+  // to let the platform know we have our listeners set up and
+  // to unblock the rendering and lookups for the rest of our application.
+  useEffect(() => {
+    Koji.ready();
+  }, []);
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <h1>My Custom Control</h1>
+        <p>Enter any text:</p>
+        <input
+          onChange={(e) => {
+            myStr = e.currentTarget.value;
+          }}
+        />
+        <button onClick={finish}>{'Save'}</button>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+
+----
+
+[.callout1]#Imports the `CustomVcc` class from the @withkoji/custom-vcc-sdk package.# +
+[.callout2]#Creates an instance of `CustomVcc`.# +
+[.callout3]#Starts event listeners to listen for updates.# +
+[.callout4]#After the customer control loads, registers it to trigger events from the parent editor.# +
+[.callout5]#When a user changes the value, updates the custom control.# +
+[.callout6]#Saves the file.# +
+[.callout7]#For testing. Displays the value returned by the custom control.#
+
+To customize the forked code for your custom control:
+
+. Modify any of the default settings and parameters.
+. Add code required by your custom control, such as:
++
+* Elements in the `return` section that users can interact with.
+* Commands in the `finish` section to perform additional processing on the captured input.
+. Clean up unneeded code.
+. When done, publish your new custom control app.
+
+Learn more: <<customvcc-reference#>>
+
+==== Provide a link to your custom control
+
+Any app that uses your custom control must call `Koji.ui.capture.custom` and provide a `koji-vccs.com` subdomain or URL that points to the custom control.
+
+* You can determine the assigned URL of the custom control.
++
+.Learn how ...
+[%collapsible]
+====
+include::{includespath}/steps-codeeditor.adoc[tag=go-to-project-details-base]
+include::{includespath}/steps-codeeditor.adoc[tag=view-published-koji]
+. Note the URL of the displayed page.
+====
+
+* You can get your own subdomain of `koji-vccs.com`.
++
+.Learn how ...
+[%collapsible]
+====
+include::{includespath}/steps-domains.adoc[tag=open-published-custom-control]
++
+include::{includespath}/steps-domains.adoc[tag=open-published-how]
+
+include::{includespath}/steps-domains.adoc[tag=menu-app-custom-domain,click-add-new-domain,select-subdomain]
+
+include::{includespath}/steps-domains.adoc[tag=create-koji-vccs-com-subdomain]
++
+This name will be passed as the `name` parameter to  `Koji.ui.capture.custom`.
+
+include::{includespath}/steps-domains.adoc[tag=click-register]
+====
+
+* You can also <<custom-domains#_using_a_domain_you_own,use a domain you own>>.
+
+==== Test your custom control
+
+. Create a standalone app.
```

Comment:
Can we provide a working code sample for the consumer app? It is super-easy to add the custom control to the React Frontend Remix Scaffold: https://withkoji.com/apps/16fb355a-6471-44df-bf62-d75a4237c13e
We can provide instructions for starting from that app and adding the few lines of code needed to consume the custom control.
That way, someone could test/debug their own code. Since the linked test app is not listed in the App Store, the reader wouldn't be able to use it that way.

Reply:
That's what the tester app is for. It's at the bottom of the section as an example. I'll move it up higher as an alternative to the React Scaffold.

Reply:
UPDATE: Still trying to create the listing. [Details in the Jira.](https://withkoji.atlassian.net/browse/DC-617?focusedCommentId=10385)

Reply:
The scaffold and example tester app are now in the App Store and added to the ["For Developers" section](https://withkoji.com/create/for-developers) by Luis.